src/rocksdb_options.rs

```diff
@@ -158,7 +158,7 @@ impl UnsafeSnap {
 
 pub struct ReadOptions {
     inner: *mut DBReadOptions,
-    upper_bound: Vec<u8>,
+    upper_bound: Option<Vec<u8>>,
```

Comment:
smaller size.
- `Option<Vec>` -> 4 usize (a pointer,  a length, a capacity, a tag)
- `Box<Vec<_>>` -> 1 usize (a pointer)
- `Option<Box<Vec>>` -> 1 usize (a pointer, since Box impls NonZero optimization)